### PR TITLE
Add @onaio/utils package

### PR DESCRIPTION
**Pull request by moshthepitt**
Fix: #133

### CODE REVIEW

**Comment by moshthepitt** [comment]
@cKellyDesign @p-netm decided not to implement `generateNameSpacedUUID` because it needs installing a dependency and at this point I do not think that it is worth it.

**Comment by p-netm** [comment]
> @cKellyDesign @p-netm decided not to implement `generateNameSpacedUUID` because it needs installing a dependency and at this point I do not think that it is worth it.

should this be like a defining rule for code, that is added to this package, that it should not have external dependencies?

**Comment by moshthepitt** [comment]
> should this be like a defining rule for code, that is added to this package, that it should not have external dependencies?

I think that any dependency should be very justified as it represents extra weight that anyone who does not use it still has to bear/carry.

cc: @cKellyDesign

File: packages/utils/src/tests/types.test.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      foo: string;
+    }
+
+    interface B {
+      bar: number;
+    }
+
+    type AB = UpdateType<A, B>;
+
+    expectType<AB>({ bar: 1337, foo: 'leet' });
```
Comment (p-netm) [inline packages/utils/src/tests/types.test.ts]: damn!! :clap: :shipit:

**Comment by moshthepitt** [comment]
@ckellyDesign

> @moshthepitt this looks nice, am wondering though if we should/could include tests where results do not match the expected type (eg var fail:Dictionary<number> = { foo: 'bar' };)?

I would really like to.  But the typescript complier and our linters will outright refuse to work with such.  I'm not sure whether the extra effort to make this work is worth it.

> Also, do any of these utils have stories?

I don't think we have anything that can have a story .... Yet.

**Comment by p-netm** [comment]
> I would really like to. But the typescript complier and our linters will outright refuse to work with such. I'm not sure whether the extra effort to make this work is worth it.

I think you could do [this](https://github.com/SamVerschueren/tsd#expectnottypetvalue)

Not sure though, testing types feels and looks different than testing code itselfs. My thoughts being that with code, only an inordinately huge amount of test permutations would sufficiently convince you the code is bug-free, but with types it seems like its the opposite; if a single happy path test passes then you are sure the type is 100% bug-free, but not necessarily strict enough or as domain specific as it should be.

I could be wrong, ofcourse.

**Comment by moshthepitt** [comment]
> I think you could do this

@p-netm @cKellyDesign just found out the existence of @ts-ignore which comes to the rescue for this.

**Comment by moshthepitt** [comment]
@cKellyDesign @p-netm I am actually going to revert the last commit.  `expectError` does not work when you also do `@ts-ignore` because it actually wants to check that an error was raised by the typescript compiler.

I am fine with not testing for negatives/bad types for now because:

- Not sure how we can make it work such that it does not blow up our linters and the typescript compiler, and I think that the extra effort to do so is not worth it right now.

**Pull request merged**